GatherPress/gatherpress
# NEW block variation "Add to calendar"

### Description of the Change



https://github.com/user-attachments/assets/7f4d1085-d300-4d17-b8e0-164359ed0b6e




Closes #606

### How to test the Change


1. Open an event
2. Add the new "Add to calendar (BUTTONS)" block.
3. Add the new "Add to calendar (DETAILS)" block.
4. Save
5. View in frontend and check that Google & Yahoo buttons work.

**Known Issue:** _ical_ & _outlook_ have no `href` attribute.

I guess that WordPress added some escaping to the sources bound to the `url` attribute of a button.

_Because_, when I make a debug-change and direct the Block Binding API to output the values into the `text` attribute, instead of `url`, one can see that the URLs are properly delivered into the block.

```json
metadata: {
	bindings: {
		text: {
			source: 'gatherpress/add-to-calendar',
			args: {
				service,
			},
		},
	},
},
```

I think the ical data-uri might just fail against `esc_url()` or something similar.

![grafik](https://github.com/user-attachments/assets/8d4346e9-f341-4097-b6bd-55a772d022e8)

I would not recommend trying to fix this or to do any changes to this PR, but instead move #928 forward!

#928 and another, yet unfinished necessary follow-up (to fully align with #831) will introduce new named endoints like `https://example.org/event/test-event-954/ical`, which will make the escaping pass and the blocks work like expected.

### Changelog Entry

> Added - New block variation "Add to calendar"

### Credits

Props @carstingaxion 

### Checklist:
